hrm-domain/lambdas/files-urls/tests/unit/getSignedS3Url.test.ts

```diff
@@ -40,6 +40,6 @@ describe('getSignedS3Url', () => {
     });
 
     const result = await getSignedS3Url(event);
-    expect(result).toEqual(newOk({ data: { media_url: mockSignedUrl } }));
+    expect(result.unwrap()).toEqual({ media_url: mockSignedUrl });
```

Comment:
@robert-bo-davis I had to change this tests because they were failing due to `unwrap` function breaking the "equal" kind of test. Changing this tests was super easy, but I'm wondering, why is `getSignedS3Url` return a `TResult` type? Shouldn't we "unwrap" it before returning that from the ALB? Or where is being consumed?

I am of the idea that the Result type should always be consumed withing the codebase of the service using it. If we want, and communicate errors across services with HTTP statuses. 

Can I ask you to help me understand what's going on here, and if adding methods to the Result type is breaking the existing behavior? Also, if that's the case and you agree with the above, can we have a short talk on how to fix this issues by consuming the Result before sending the response to the ALB?

Reply:
The [alb-handler package](https://github.com/techmatters/hrm/blob/master/lambdas/packages/alb-handler/index.ts) handles converting the Result to an appropriate http status. The [getSignedS3Url function isn't returned directly](https://github.com/techmatters/hrm/blob/master/hrm-domain/lambdas/files-urls/index.ts#L30) as part of the ALB response.